### PR TITLE
Add json merge support

### PR DESCRIPTION
**Pull request by k0nserv**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -255,6 +255,18 @@ where
                 ..Default::default()
             };
 
+            // TODO: Explore using `tracing`'s `Valuable` trait instead of this when it becomes
+            // stable.
+            if let Some(Value::String(raw_json)) = json.remove("json") {
+                let parse_result = serde_json::from_str(&raw_json);
```
Comment (lolgesten) [inline src/lib.rs:261]: This seems very expensive to do on every metrics logging.

If it isn't valid JSON, we will just get a parse error in log ingester, and the row discarded – I think that's enough protection.

Comment (k0nserv) [reply src/lib.rs:261]: I suppose, that's very hacky though. My thinking is that we are already doing a synchronous HTTP call in this code path and this parsing cost feels negligible in comparison. I would think the HTTP call would dominate

Comment (lolgesten) [reply src/lib.rs:261]: alright

Comment (lolgesten) [reply src/lib.rs:261]: there is no HTTP call.

Comment (k0nserv) [reply src/lib.rs:261]: I meant in `handle_log_record` which is called at the end of this function

Comment (lolgesten) [reply src/lib.rs:261]: yeah. still not HTTP. it's just writing to a socket. which in turn will be writing to a buffer in the OS. it's very fast.

Comment (lolgesten) [reply src/lib.rs:261]: Uhm. What do you reckon the bandwidth is between two adjacent EC2 hosts? MB/s GB/s?

In what reality would we be logging even close to the bandwidth available?

Comment (k0nserv) [reply src/lib.rs:261]: Yeah you are probably right that it's not going to be a problem. Still the second point about WellKnown being `Serialize` remains

Comment (lolgesten) [reply src/lib.rs:261]: alright

**Review by lolgesten — APPROVED** [review]

**Pull request merged**